homework/index.js

```diff
@@ -29,18 +29,78 @@
     });
     return elem;
   }
+  function renderRepositories(repository, container) {
+    container.innerHTML = '';
+    const table = createAndAppend('table', container);
+    const tbody = createAndAppend('tbody', table);
+    const tr1 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr1, { text: 'Repository:' });
+    const td12 = createAndAppend('td', tr1);
+    createAndAppend('a', td12, {
+      href: repository.html_url,
+      target: '_blank',
+      text: repository.name,
+    });
+    const tr2 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr2, { text: 'Description:' });
+    createAndAppend('td', tr2, { text: repository.description });
+    const tr3 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr3, { text: 'Forks:' });
+    createAndAppend('td', tr3, { text: repository.forks });
+    const tr4 = createAndAppend('tr', tbody);
+    createAndAppend('td', tr4, { text: 'Updated:' });
+    createAndAppend('td', tr4, { text: repository.updated_at });
+  }
+
+  function renderContributors(contributors, container) {
+    container.innerHTML = '';
+    createAndAppend('p', container, { text: 'Contributions' });
+    const ul = createAndAppend('ul', container);
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul);
+      const a = createAndAppend('a', li, {
+        href: contributor.html_url,
+        target: '_blank',
+        text: contributor.login,
+      });
+      createAndAppend('img', a, { src: contributor.avatar_url, width: 100, height: 100 });
+      createAndAppend('div', a, { text: contributor.contributions });
+    });
+  }
 
   function main(url) {
-    fetchJSON(url, (err, data) => {
+    fetchJSON(url, (err, repositories) => {
       const root = document.getElementById('root');
+      const header = createAndAppend('div', root, { id: 'header' });
+      createAndAppend('label', header, { text: 'HYF Repositories' });
+      const select = createAndAppend('select', header);
+
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        repositories.sort((x, y) => x.name.localeCompare(y.name));
+        const leftHand = createAndAppend('div', root, { id: 'left-hand' });
+        const rightHand = createAndAppend('div', root, { id: 'right-hand' });
+
+        repositories.forEach((repository, index) => {
+          createAndAppend('option', select, { text: repository.name, value: index });
+        });
+        select.addEventListener('change', () => {
+          const index = select.value;
+          const repository = repositories[index];
+          renderRepositories(repository, leftHand);
+          fetchJSON(repository.contributors_url, (_err, contributors) => {
```

Comment:
Small thing here - these lines (92 - 94) are almost identical to lines 98 - 100. Perhaps extract into a function.